OpenLaTeXStudio/EditorModule/src/latexstudio/editor/menu/OpenTemplate.java

```diff
@@ -0,0 +1,121 @@
+package latexstudio.editor.menu;
+
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.swing.DefaultListModel;
+import javax.swing.JDialog;
+import org.w3c.dom.Document;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import latexstudio.editor.EditorTopComponent;
+import latexstudio.editor.TopComponentFactory;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Text;
+import org.xml.sax.SAXException;
+
+@ActionID(
+        category = "File",
+        id = "latexstudio.editor.OpenTemplate"
+)
+@ActionRegistration(
+        displayName = "#CTL_OpenTemplate"
+)
+@ActionReference(path = "Menu/File", position = 1201)
+@NbBundle.Messages("CTL_OpenTemplate=Open Template")
+
+public final class OpenTemplate extends JDialog implements ActionListener {
+
+    private static final String pathToTemplatesDirectory = "/latexstudio/editor/resources/templates/";
+
+    private final EditorTopComponent etc = new TopComponentFactory<EditorTopComponent>()
+            .getTopComponent(EditorTopComponent.class.getSimpleName());
+
+    private OpenTemplate() {
+        //Dialog window properties
+        setTitle("Choose Template");
+        setLocationRelativeTo(etc);
+        setLayout(new BorderLayout(10, 10));
+        setResizable(false);
+        setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
+        setLayout(new BorderLayout(10, 10));
+        setModalityType(ModalityType.APPLICATION_MODAL);
+
+        add(new OpenTemplatePanel(this, getListModelWithTemplates(readXML())), BorderLayout.CENTER);
+        pack();
+    }
+
+    /**
+     @return Array of nodes that contain informations about preinstalled
+     templates
+     */
+    private ArrayList<Element> readXML() {
+        try {
+            Document templatesXML = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(getClass().getResourceAsStream("/latexstudio/editor/resources/templates.xml"));
+
+            Element rootElement = templatesXML.getDocumentElement();
+            NodeList templatesNodeList = rootElement.getChildNodes();
+
+            ArrayList<Element> templates = new ArrayList<Element>();
+            for (int i = 0; i < templatesNodeList.getLength(); i++) {
```

Comment:
How I can do that? NodeList doesn't implements Iterable interface (and it's also an interface, so it can't implement another).